src/app/AttributePathExpandIterator.h

```diff
@@ -76,8 +77,24 @@ class AttributePathExpandIterator
         static Position StartIterating(SingleLinkedListNode<AttributePathParams> * path) { return Position(path); }
 
         /// Copies are allowed
-        Position(const Position &)             = default;
-        Position & operator=(const Position &) = default;
+        Position(const Position &) = default;
+        Position & operator=(const Position & other)
+        {
+            if (this != &other)
+            {
+                mAttributePath = other.mAttributePath;
+                mOutputPath    = other.mOutputPath;
+                if (other.mEntry)
+                {
+                    mEntry.emplace(*other.mEntry);
+                }
+                else
+                {
+                    mEntry.reset();
+                }
```

Comment:
Again, this seems like a really long way to write `mEntry = other.mEntry;`.

Reply:
Entires have a const field so they are not assignable. However in the mean time I removed this as it was increasing RAM.